Makefile

```diff
@@ -0,0 +1,164 @@
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DOCKER   ?= docker
+MKDIR    ?= mkdir
+TR       ?= tr
+DIST_DIR ?= $(CURDIR)/dist
+
+include $(CURDIR)/common.mk
+
+BUILDIMAGE_TAG ?= golang$(GOLANG_VERSION)
+BUILDIMAGE ?= $(IMAGE_NAME)-build:$(BUILDIMAGE_TAG)
+
+CMDS := $(patsubst ./cmd/%/,%,$(sort $(dir $(wildcard ./cmd/*/))))
+CMD_TARGETS := $(patsubst %,cmd-%, $(CMDS))
+
+CHECK_TARGETS := assert-fmt vet lint ineffassign misspell
+MAKE_TARGETS := binaries build check fmt lint-internal test examples cmds coverage generate $(CHECK_TARGETS)
+
+TARGETS := $(MAKE_TARGETS) $(CMD_TARGETS)
+
+DOCKER_TARGETS := $(patsubst %,docker-%, $(TARGETS))
+.PHONY: $(TARGETS) $(DOCKER_TARGETS)
+
+GOOS ?= linux
+
+binaries: cmds
+ifneq ($(PREFIX),)
+cmd-%: COMMAND_BUILD_OPTIONS = -o $(PREFIX)/$(*)
+endif
+cmds: $(CMD_TARGETS)
+$(CMD_TARGETS): cmd-%:
+	CGO_LDFLAGS_ALLOW='-Wl,--unresolved-symbols=ignore-in-object-files' GOOS=$(GOOS) \
+		go build -ldflags "-s -w -X main.version=$(VERSION)" $(COMMAND_BUILD_OPTIONS) $(MODULE)/cmd/$(*)
+
+build:
+	GOOS=$(GOOS) go build ./...
+
+examples: $(EXAMPLE_TARGETS)
+$(EXAMPLE_TARGETS): example-%:
+	GOOS=$(GOOS) go build ./examples/$(*)
+
+all: check test build binary
+check: $(CHECK_TARGETS)
```

Comment:
Personally I am a fan of golangci-lint. But this is bike-shedding territory, we can merge this as-is and iterate later.

Reply:
For comparison, here's what golangci-lint with its default config reports:

```
cmd/set-nas-status/main.go:79:20: Error return value of `(*github.com/spf13/pflag.FlagSet).Set` is not checked (errcheck)
				cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val))
				               ^
cmd/dra-example-kubeletplugin/driver.go:104:16: Error return value of `d.state.Free` is not checked (errcheck)
			d.state.Free(req.ClaimUid)
			            ^
cmd/dra-example-kubeletplugin/main.go:93:20: Error return value of `(*github.com/spf13/pflag.FlagSet).Set` is not checked (errcheck)
				cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val))
				               ^
cmd/dra-example-kubeletplugin/main.go:236:17: Error return value of `driver.Shutdown` is not checked (errcheck)
	driver.Shutdown()
	               ^
cmd/dra-example-controller/main.go:110:20: Error return value of `(*github.com/spf13/pflag.FlagSet).Set` is not checked (errcheck)
				cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val))
				               ^
cmd/dra-example-controller/requests.go:46:2: S1008: should use 'return exists' instead of 'if !exists { return false }; return true' (gosimple)
	if !exists {
	^
cmd/dra-example-kubeletplugin/main.go:213:10: printf: fmt.Errorf format %v reads arg #1, but call has 0 args (govet)
		return fmt.Errorf("path for cdi file generation is not a directory: '%v'")
		       ^
```

Reply:
Updated. I've actually been trying to find a good replacement for `golint` since it's no longer officially maintained.

Reply:
Let me know if the change here looks good before I resolve this.

Reply:
@klueska it fails in my env:
```
$ make check
...
go list -f '{{.Dir}}' github.com/kubernetes-sigs/dra-example-driver/... | xargs golangci-lint run
xargs: golangci-lint: terminated by signal 9
Makefile:83: recipe for target 'lint' failed
make: *** [lint] Error 125
```

Reply:
It will fail if you don't have `golangci-lint` installed. Try running `make docker-lint` and that should work for you. all targets can be ran with `docker-*` to run them inside docker.

Reply:
As mentioned [here](https://github.com/kubernetes-sigs/dra-example-driver/pull/1#discussion_r1151766420), this appears to be a peculiarity of @bart0sh's environment. It will not block this PR.